components/core/src/streaming_compression/passthrough/Decompressor.cpp

```diff
@@ -5,51 +5,90 @@
 
 namespace streaming_compression { namespace passthrough {
     ErrorCode Decompressor::try_read (char* buf, size_t num_bytes_to_read, size_t& num_bytes_read) {
-        if (nullptr == m_compressed_data_buf) {
-            return ErrorCode_NotInit;
+        if (InputType::NotInitialized == m_input_type) {
```

Comment:
it might be helpful to have a few comments so reader can tell what's the criteria to decide if an error should be throw

Reply:
My criteria is if it's a logic error, I'll throw an exception. It may be a bit redundant to add that comment across the codebase. How about I add it as a coding guideline later?

Reply:
gotcha, that makes sense